core/src/main/java/org/jboss/pnc/build/finder/core/DistributionAnalyzer.java

```diff
@@ -318,6 +327,10 @@ private static boolean isJar(FileObject fo) {
         return Collections.unmodifiableMap(map);
     }
 
+    private String getCacheLocation() {
+        return new File(ConfigDefaults.CONFIG_PATH, "cache").getAbsolutePath();
```

Comment:
OK, I guess you got this from `Main.initCaches` which has `String location = new File(ConfigDefaults.CONFIG_PATH, "cache").getAbsolutePath();`.

But, here is not the right place for it. We need the actual value it has been set to.

Then we have `Main` call something to set it, and only return the default path if unset.

Also, we could define a global default in `ConfigDefaults`, but defaulting to `CONFIG_PATH, "cache"` is OK, too.

Reply:
I've added an option to specify cache location so that we take the value from config instead of creating it on the spot. Is this a suitable solution? :)

Reply:
The current value appears to be `ConfigDefaults.CONFIG_PATH + "/cache"`.

Rather than specify the cache directory, we should just specify on the command line the default app home directory, which is normally `${user.home}/.build-finder` and allow the user to change this location, but then the cache path would always be relative to that location. What do you think?

Reply:
Yeah, I was initially against having the option to specify the cache-location because I didn't like the implementation but it is not that ugly I suppose. Keeping it relative to the app-directory is better, I agree.

Reply:
Yeah, I like your proposal much more. :)

Reply:
Is it possible to change the cache location now? Looking at the code, I wasn't sure, the `ConfigDefaults` file was changed, but these paths should be immutable.

Reply:
No, it shouldn't be possible to change cache location. The cache location is bound to the root path (app home directory), which is unchangeable.